src/OpenSpaceToolkit/Astrodynamics/EventCondition/SemiMajorAxisCondition.cpp

```diff
@@ -0,0 +1,57 @@
+/// Apache License 2.0
+
+#include <OpenSpaceToolkit/Astrodynamics/EventCondition/SemiMajorAxisCondition.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Kepler/COE.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace eventcondition
+{
+
+using ostk::physics::units::Derived;
+using ostk::physics::units::Length;
+using ostk::physics::units::Time;
+
+using ostk::astro::trajectory::orbit::models::kepler::COE;
+
+static const Derived::Unit GravitationalParameterSIUnit =
+    Derived::Unit::GravitationalParameter(Length::Unit::Meter, Time::Unit::Second);
+
+SemiMajorAxisCondition::SemiMajorAxisCondition(
+    const Criteria& aCriteria, const Length& aLength, const Derived& aGravitationalParameter
+)
+    : EventCondition("Semi-Major Axis Condition", aCriteria),
+      smaInMeters_(aLength.inMeters()),
+      gravitationalParameterInSI_(aGravitationalParameter.in(GravitationalParameterSIUnit))
+{
+}
+
+SemiMajorAxisCondition::~SemiMajorAxisCondition() {}
+
+Length SemiMajorAxisCondition::getSemiMajorAxis() const
+{
+    return Length::Meters(smaInMeters_);
+}
+
+Derived SemiMajorAxisCondition::getGravitationalParameter() const
+{
+    return {gravitationalParameterInSI_, GravitationalParameterSIUnit};
+}
+
+Real SemiMajorAxisCondition::evaluate(const VectorXd& aStateVector, const Real& aTime) const
```

Comment:
This is what I meant about having to make some assumptions when operating one level below `State`

We are making 2 implicit assumptions here. Is going to work for us 99.9% of the time, but still I think if we can avoid them, it's a better dev experience. For instance, assumptions here are:

1) VectorXd coordinates are expressed w.r.t. an inertial frame (it is likely that this is always the case)
2) VectorXd coordinates are expressed w.r.t. a frame centered at the central body (it is likely that this is always the case)

Still, to me the robust way to go about it:
1. Provide not only the standard gravitational parameter in the constructor but also the inertial frame of choice centered at the desired central body (ideally we could only pass the `Celestial` body but I dont think it offers a `getInertialFrame`.
2. In the `evaluate` logic, you receive a `State`, extract pos/vel from it in the frame specified in the constructor, and then proceed to compute the semi-major axis.

As I said, not that it will fail, but I see this as a risk.
Sorry being a pain in the ass again.

Reply:
Not at all, this is great feedback. I think we have to find the balance between using high level objects and low level data structures. As we discussed offline, we are going to reassess this once we land your MR on the Coordinate Broker.